src/Mvc/Mvc.Core/src/Infrastructure/IActionResultExecutor.cs

```diff
@@ -16,7 +18,7 @@ namespace Microsoft.AspNetCore.Mvc.Infrastructure
     /// <see cref="IActionResult.ExecuteResultAsync(ActionContext)"/> method of the corresponding action result type.
     /// Implementations should be registered as singleton services.
     /// </remarks>
-    public interface IActionResultExecutor<in TResult> where TResult : IActionResult
+    public interface IActionResultExecutor<in TResult> where TResult : notnull, IActionResult
```

Comment:
Is notnull a breaking change constraint?

Reply:
It's a source breaking change in some cases. However in this particular case, MVC does not invoke this API when the action result is null, so it's kinda moot.